### ISSUE TITLE
[Next] [Typescript] creating an interface from styled component raises an error

### ISSUE THREAD

**Issue by itsramiel** [marklawlor/nativewind#332]
**Describe the bug**
Usually when I create a custom component, I would like to be able to pass all props that the container component can accept. Usually I do this by 
```

const ContainerComponent = styled(View, "py-3")

interface CustomComponentProps extends ComponentProps<typeof ContainerComponent> {}

const CustomComponent = (props: CustomComponentProps) => {
    return <ContainerComponent {...props} />
}
```

This code works if you do not use a styled nativewind component, but with nativewind I get the following error when defining the interface: 
<img width="1012" alt="image" src="https://user-images.githubusercontent.com/80689446/202555115-fa92032f-a22f-4f6d-a8f1-4f5bbb7792ee.png">

**To Reproduce**
Follow the code above or use the repro below

**Expected behavior**
I expect the type to not error

**Repro**
https://github.com/itsramiel/nativewind-pressable/tree/repro/custom-component

**Comment by marklawlor** [comment]
Fixed in `3.0.0-next.22`

**Comment by itsramiel** [comment]
I appreciate your work on this library!
